File: examples/java/core/wait/named-wait.java
import java.time.Duration;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class NamedWaitExample extends DurableHandler<Object, String> {
    @Override
    public String handleRequest(Object input, DurableContext context) {
        // Wait with explicit name
        context.wait("custom_wait", Duration.ofSeconds(2));
        return "Wait with name completed";
    }
}

File: examples/csharp/core/wait/named-wait.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;

public class NamedWaitExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, string>(Workflow, input, context);

    private async Task<string> Workflow(object input, IDurableContext ctx)
    {
        // Wait with explicit name
        await ctx.WaitAsync(TimeSpan.FromSeconds(2), name: "custom_wait");
        return "Wait with name completed";
    }
}
